GCDWebServer/Responses/GCDWebServerFileResponse.m

```diff
@@ -88,26 +88,30 @@ - (instancetype)initWithFile:(NSString*)path byteRange:(NSRange)range isAttachme
     return nil;
   }
 #endif
-  NSUInteger fileSize = (NSUInteger)info.st_size;
+  
+  if ((self = [super init])) {
+    
+    NSUInteger fileSize = (NSUInteger)info.st_size;
 
-  BOOL hasByteRange = GCDWebServerIsValidByteRange(range);
-  if (hasByteRange) {
-    if (range.location != NSUIntegerMax) {
-      range.location = MIN(range.location, fileSize);
-      range.length = MIN(range.length, fileSize - range.location);
+    BOOL hasByteRange = GCDWebServerIsValidByteRange(range);
+    if (hasByteRange) {
+      if (range.location != NSUIntegerMax) {
+        range.location = MIN(range.location, fileSize);
+        range.length = MIN(range.length, fileSize - range.location);
+      } else {
+        range.length = MIN(range.length, fileSize);
+        range.location = fileSize - range.length;
+      }
+      if (range.length == 0) {
+        [self setStatusCode:kGCDWebServerHTTPStatusCode_RequestedRangeNotSatisfiable];
+        [self setValue:[NSString stringWithFormat:@"bytes */%lu", (unsigned long)fileSize] forAdditionalHeader:@"Content-Range"];
+        return self;
```

Comment:
I don't think this will work correctly: you are returning a partially initialized `GCDWebServerFileResponse`. The reason for this TODO is that you would want to return a different response class instance, which is not trivial.